<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>Manipulation de données et production de figures sous R</title>

		<meta name="description" content="dplyR, tidyR et ggplot2">
		<meta name="author" content="collectif de bioinformatique de Baillarguet">

		<meta name="apple-mobile-web-app-capable" content="yes">
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">

		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">

		<link rel="stylesheet" href="css/reveal.css">
		<link rel="stylesheet" href="css/theme/solarized.css" id="theme">

		<!-- Code syntax highlighting -->
		<link rel="stylesheet" href="lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
		
		
		<script src="lib/htmlwidgets-0.5/htmlwidgets.js"></script>
		<script src="lib/plotlyjs-1.1.1/plotly-latest.min.js"></script>
		<script src="lib/plotly-binding-2.0.6/plotly.js"></script>

		
		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>

	  <div class="reveal">
		<div class="slides">
          
          <!-- Titre -->
		  <section>
			<h1>Manipulation de données et visualisation sous R</h1>
			<h3>ggplot2, dyplr & tidyr</h3>
			<p>
			  <small>collectif de bioinformatique de Baillarguet</small>
			</p>
		  </section>

		  <!-- Historique de R -->
		  <section>
			<h2>R</h2>
			<a href="https://www.r-project.org/">
              <img data-src="https://www.r-project.org/Rlogo.png" alt="R logo">
            </a>
            <ul>
			  <li>1993</li>
			  <li>logiciel libre</li>
			  <li>multi-plateforme</li>
              <li>modulaire et en forte croissance</li>
			</ul>
		  </section>

		  <!-- Popularité de R -->
		  <section>
			<h2>Popularité croissante</h2>
			<a href="http://r4stats.com/articles/popularity/">
              <img data-src="http://i1.wp.com/datasciencepopularity.com/wp-content/uploads/2015/05/fig_2f_scholarlyimpactallstat21.png" alt="R's popularity">
            </a>
		  </section>

		  <!-- Nombre de modules -->
		  <section>
			<h2>Nombre de modules croissant</h2>
			<a href="http://r4stats.com/articles/popularity/">
              <img
                 width="500"
                 data-src="http://i1.wp.com/r4stats.com/wp-content/uploads/2015/05/fig_9_cran.png"
                 alt="R's modules">
            </a>
		  </section>

		  <!-- Hadley Wickham -->
		  <section>
			<h2>Hadley Wickham</h2>
            <a href="https://www.youtube.com/watch?v=LOXe6Eu59As">
              <img
                 data-src="https://i.ytimg.com/vi/LOXe6Eu59As/maxresdefault.jpg"
                 alt="Hadley Wickham">
            </a>
		  </section>

		  <!-- Hadley Wickham modules -->
		  <section>
			<h2>Hadley Wickham</h2>
			<h3>manipulation (unifiée) et visualisation de données</h3>
            <ul>
			  <li><a href="https://cran.r-project.org/web/packages/stringr/index.html">stringr</a></li>
			  <li><a href="https://cran.r-project.org/web/packages/dplyr/index.html">dplyr</a></li>
			  <li><a href="https://cran.r-project.org/web/packages/tidyr/index.html">tidyr</a></li>
			  <li><a href="http://ggplot2.org/">ggplot2</a> (<a href="https://www.springer.com/us/book/9780387245447">grammar of graphics</a> par Leland Wilkinson)</li>
			  <li><a href="http://ggvis.rstudio.com/">ggvis</a></li>
			  <li>Chief Scientist <a href="https://www.rstudio.com/">RStudio</a></li>
			</ul>
		  </section>

		  <!-- RStudio -->
		  <section>
			<h2>RStudio</h2>
			<h3>interface graphique pour R (et bien plus)</h3>
            <a href="https://www.rstudio.com/">
              <img width="400"
                   data-src="https://www.rstudio.com/wp-content/uploads/2014/06/RStudio-Ball.png"
                   >
            </a>
		  </section>

		  <!-- Première démonstration : données simples -->
		  <section>
			<h2>Cleaning and visualizing genomic data</h2>
			<h3>a case study in tidy analysis</h3>
            <a href="http://varianceexplained.org/r/tidy-genomics/">
              <img width="200"
                   data-src="http://varianceexplained.org/images/david_robinson_picture2.jpg"
                   >
            </a>
            <p>par David Robinson</p>
		  </section>

		  <!-- Source des données -->
		  <section>
			<h2>Coordination of Growth Rate, Cell Cycle, Stress Response, and Metabolic Activity in Yeast</h2>
			<h3>Brauer et al. (2008) Mol. Biol. Cell January 1, 2008 vol. 19 no. 1 352-367 (doi: 10.1091/mbc.E07-08-0779)</h3>
            <p>http://varianceexplained.org/files/Brauer2008_DataSet1.tds</p>
		  </section>

		  <!-- Premières commandes avec RStudio -->
		  <section>
			<h2>Premières commandes avec RStudio</h2>
			<pre>
              <code class="R" data-trim contenteditable>
Brauer2008_DataSet1 <- read.delim("/tmp/Rtmp3mlo42/data1aaa3ddc8686")
View(Brauer2008_DataSet1)                
dim(Brauer2008_DataSet1)
			  </code>
            </pre>
			<p>Nombre de colonnes et de lignes</p>
		  </section>

		  <!-- Description des données -->
		  <section>
			<h2>Description des données</h2>
            <p>6 milieux de culture en déficit d'un nutriment :</p>
            <ul>
			  <li>glucose (G)</li>
			  <li>ammonium (N)</li>
			  <li>phosphate (P)</li>
			  <li>sulfate (S)</li>
			  <li>leucine (L)</li>
			  <li>uracil (U)</li>
			</ul>
            <p>6 vitesses de croissance: lent (0,05) à rapide (0,3).</p>
		  </section>

		  <!-- Pourquoi changer la disposition des données ? -->
		  <section>
			<h2>Disposition des données : pourquoi changer ?</h2>
            <p>Les outils d'Hadley Wickham réclament :</p>
            <ul>
			  <li>chaque variable forme une colonne,</li>
			  <li>chaque observation forme une ligne,</li>
			  <li>un type d'observation par tableau.</li>
			</ul>
            <p>Ce qui ne va pas dans la disposition actuelle :</p>
            <ul>
			  <li class="fragment">G0.05 n'est pas un nom de variable,</li>
			  <li class="fragment">plusieurs variables dans la colonne NAME.</li>
			</ul>
		  </section>

		  <!-- Traitement de la colonne NAME ? -->
		  <section>
			<h2>Séparer la colonne NAME en plusieurs variables</h2>
            <p>La colonne NAME contient :</p>
            <ul>
			  <li>un nom de gène (ex : SFB2) ou rien,</li>
			  <li>un nom de processus biologique,</li>
			  <li>une fonction moléculaire,</li>
			  <li>un identifiant,</li>
			  <li>un autre identifiant (?)</li>
			</ul>
            <p>Séparer les variables facilite l'exploration et permet de poser des questions plus précises.</p>
		  </section>

		  <!-- Utilisation de dplyr et tidyr -->
		  <section>
		    <section>
			  <h2>Séparer la colonne NAME en plusieurs variables</h2>
			  <pre>
                <code class="R" data-trim>
library(dplyr)
library(tidyr)
cleaned_data <- original_data %>%
    separate(NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|")
View(cleaned_data)
			    </code>
              </pre>
		    </section>
		    <section>
			  <h2>Autre possibilité</h2>
			  <pre>
                <code class="R" data-trim>
library(dplyr)
library(tidyr)
cleaned_data <- separate(Brauer2008_DataSet1, NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|")
			    </code>
              </pre>
		    </section>
		  </section>

		  <!-- retrait des espaces -->
		    <section>
			  <h2>Élimination des espaces</h2>
			  <pre>
                <code class="R" data-trim>
head(cleaned_data$BP)
			    </code>
              </pre>
			  <pre>
## [1] " ER to Golgi transport "        " biological process unknown "  
## [3] " proteolysis and peptidolysis " " mRNA polyadenylylation* "     
## [5] " vesicle fusion* "              " biological process unknown "
              </pre>
		    </section>

		  <!-- retrait des espaces -->
		    <section>
			  <h2>Élimination des espaces</h2>
              <p>fonction trimws (<i>trim whitespace</i>) :</p>
			  <pre>
                <code class="R" data-trim>
cleaned_data <- Brauer2008_DataSet1 %>%
    separate(NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|") %>%
    mutate_each(funs(trimws), name:systematic_name)
			    </code>
              </pre>
              <pre>
[1] "ER to Golgi transport"        "biological process unknown"  
[3] "proteolysis and peptidolysis" "mRNA polyadenylylation*"     
[5] "vesicle fusion*"              "biological process unknown"
              </pre>
		    </section>

		  <!-- élimination de colonnes -->
		    <section>
			  <h2>Élimination de colonnes</h2>
              <p>fonction <i>select</i> de dplyr :</p>
			  <pre>
                <code class="R" data-trim>
cleaned_data <- Brauer2008_DataSet1 %>%
    separate(NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|") %>%
    mutate_each(funs(trimws), name:systematic_name) %>%
    select(-number, -GID, -YORF, -GWEIGHT)
glimpse(cleaned_data)
			    </code>
              </pre>
              <pre>
Observations: 5537
Variables:
$ name            (chr) "SFB2", "", "QRI7", "CFT2", "SSO2", "PSP2", ...
$ BP              (chr) "ER to Golgi transport", "biological process...
$ MF              (chr) "molecular function unknown", "molecular fun...
$ systematic_name (chr) "YNL049C", "YNL095C", "YDL104C", "YLR115W", ...
$ G0.05           (dbl) -0.24, 0.28, -0.02, -0.33, 0.05, -0.69, -0.5...
$ G0.1            (dbl) -0.13, 0.13, -0.27, -0.41, 0.02, -0.03, -0.3...
$ G0.15           (dbl) -0.21, -0.40, -0.27, -0.24, 0.40, 0.23, -0.1...
...
              </pre>
		    </section>


		  <!-- Extraire les variables des noms de colonnes -->
		    <section>
			  <h2>Passage au format Wickham</h2>
              <p>objectif : une colonne par variable. Passage de 36
              colonnes à 3 (nutriment, taux de croissance, expression) avec la
              fonction <i>gather</i> de tidyr :</p>
			  <pre>
                <code class="R" data-trim>
cleaned_data <- Brauer2008_DataSet1 %>%
    separate(NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|") %>%
    mutate_each(funs(trimws), name:systematic_name) %>%
    select(-number, -GID, -YORF, -GWEIGHT) %>%
    gather(sample, expression, G0.05:U0.3)
glimpse(cleaned_data)
			    </code>
              </pre>
		    </section>

		  <!-- Extraire les variables des noms de colonnes -->
		    <section>
			  <h2>Passage au format Wickham</h2>
              <p>objectif : une colonne par variable. Passage de 36
              colonnes à 3 (nutriment, taux de croissance, expression) avec la
              fonction <i>gather</i> de tidyr :</p>
              <pre>
Observations: 199332
Variables:
$ name            (chr) "SFB2", "", "QRI7", "CFT2", "SSO2", "PSP2", ...
$ BP              (chr) "ER to Golgi transport", "biological process...
$ MF              (chr) "molecular function unknown", "molecular fun...
$ systematic_name (chr) "YNL049C", "YNL095C", "YDL104C", "YLR115W", ...
$ sample          (fctr) G0.05, G0.05, G0.05, G0.05, G0.05, G0.05, G...
$ expression      (dbl) -0.24, 0.28, -0.02, -0.33, 0.05, -0.69, -0.5...
...
              </pre>
		    </section>

		  <!-- Extraire les variables des noms de colonnes -->
		    <section>
			  <h2>Séparation des variables nutriment et taux de croissance</h2>
              <p>fonction <i>separate</i> de tidyr :</p>
			  <pre>
                <code class="R" data-trim>
cleaned_data <- Brauer2008_DataSet1 %>%
    separate(NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|") %>%
    mutate_each(funs(trimws), name:systematic_name) %>%
    select(-number, -GID, -YORF, -GWEIGHT) %>%
    gather(sample, expression, G0.05:U0.3) %>%
    separate(sample, c("nutrient", "rate"), sep = 1, convert = TRUE)
glimpse(cleaned_data)
			    </code>
			  </pre>
		    </section>

		  <!-- Extraire les variables des noms de colonnes -->
		    <section>
			  <h2>Séparation des variables nutriment et taux de croissance</h2>
              <p>fonction <i>separate</i> de tidyr :</p>
			  <pre>
Observations: 199332
Variables:
$ name            (chr) "SFB2", "", "QRI7", "CFT2", "SSO2", "PSP2", ...
$ BP              (chr) "ER to Golgi transport", "biological process...
$ MF              (chr) "molecular function unknown", "molecular fun...
$ systematic_name (chr) "YNL049C", "YNL095C", "YDL104C", "YLR115W", ...
$ nutrient        (fctr) G, G, G, G, G, G, G, G, G, G, G, G, G, G, G...
$ rate            (dbl) 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0....
$ expression      (dbl) -0.24, 0.28, -0.02, -0.33, 0.05, -0.69, -0.5...
			  </pre>
		    </section>

		  <!-- Selection d'un gène d'intérêt -->
		    <section>
			  <h2>Focus sur le gène LEU1</h2>
              <p>fonction <i>filter</i> de dplyr :</p>
			  <pre>
                <code class="R" data-trim>
cleaned_data %>%
  filter(name == "LEU1") %>%
  View()
			    </code>
			  </pre>
              <p>LEU1 est un gène participant à la synthèse de la
              leucine. On a 36 observations (6 conditions et 6 taux de
              croissance).</p>
		    </section>

		  <!-- Selection d'un gène d'intérêt -->
		  <section>
		    <section>
			  <h2>Visualisation avec ggplot2</h2>
              <p>fonction <i>filter</i> de dplyr :</p>
			  <pre>
                <code class="R" data-trim>
library(ggplot2)

cleaned_data %>%
  filter(name == "LEU1") %>%
  ggplot(aes(rate, expression, color = nutrient)) +
  geom_line()
			    </code>
			  </pre>
              <p>LEU1 est un gène participant à la synthèse de la
              leucine. On a 36 observations (6 conditions et 6 taux de
              croissance).</p>
		    </section>
		    <section>
			  <h2>ggplot2 ? gnéé ??</h2>
              <p>grammaire graphique :</br>données + géométrie + système de coordonnées = graphique</p>
			  <pre>
                <code class="R" data-trim>
  ggplot(data = cleaned_data,
         aes(x = rate, y = expression, color = nutrient)) +
  geom_line()
			    </code>
			  </pre>
              <p>système de coordonnées cartésien par défaut (polaire,
              logarithmique, cartographique, ...)</p>
		    </section>
		  </section>

		  <!-- Selection d'un gène d'intérêt -->
		    <section>
			  <h2>Visualisation avec ggplot2</h2>
              <img
                 data-src="http://varianceexplained.org/figs/2015-11-19-tidy-genomics/unnamed-chunk-14-1.png"
                 >
		    </section>

		  <!-- Selection de plusieurs gènes d'intérêt -->
		    <section>
			  <h2>Focus sur les gènes LEU</h2>
			  <pre>
                <code class="R" data-trim>
cleaned_data %>%
  filter(BP == "leucine biosynthesis") %>%
  ggplot(aes(rate, expression, color = nutrient)) +
  geom_line() +
  facet_wrap(~name)
			    </code>
			  </pre>
		    </section>

		  <!-- Selection d'un gène d'intérêt -->
		    <section>
			  <h2>Visualisation avec ggplot2</h2>
              <img
                 data-src="http://varianceexplained.org/figs/2015-11-19-tidy-genomics/unnamed-chunk-15-1.png"
                 >
		    </section>

		  <!-- Selection de plusieurs gènes d'intérêt -->
		    <section>
			  <h2>Focus sur les gènes LEU</h2>
              <p>ajustement de droite (modèle linéaire, ou <i>lm</i>)</p>
			  <pre>
                <code class="R" data-trim>
cleaned_data %>%
  filter(BP == "leucine biosynthesis") %>%
  ggplot(aes(rate, expression, color = nutrient)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~name)
			    </code>
			  </pre>
		    </section>

		  <!-- Selection d'un gène d'intérêt -->
		    <section>
			  <h2>Visualisation avec ggplot2</h2>
              <img
                 data-src="http://varianceexplained.org/figs/2015-11-19-tidy-genomics/unnamed-chunk-16-1.png"
                 >
		    </section>

		  <!-- Selection de métabolisme d'intérêt -->
		    <section>
			  <h2>Focus sur le métabolisme du soufre</h2>
			  <pre>
                <code class="R" data-trim>
cleaned_data %>%
  filter(BP == "sulfur metabolism") %>%
  ggplot(aes(rate, expression, color = nutrient)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~name + systematic_name, scales = "free_y")
			    </code>
			  </pre>
		    </section>

		  <!-- Selection d'un gène d'intérêt -->
		    <section>
			  <h2>Visualisation avec ggplot2</h2>
              <img
                 data-src="http://varianceexplained.org/figs/2015-11-19-tidy-genomics/unnamed-chunk-17-1.png"
                 >
		    </section>

		  <!-- Toutes les commandes -->
		  <section>
			<h2>Script complet</h2>
			<pre>
              <code class="R" data-trim>
library(dplyr)
library(tidyr)
library(ggplot2)

Brauer2008_DataSet1 <- read.delim("/tmp/RtmpmIvzA5/data67167154fe1d")

cleaned_data <- Brauer2008_DataSet1 %>%
  separate(NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|") %>%
  mutate_each(funs(trimws), name:systematic_name) %>%
  select(-number, -GID, -YORF, -GWEIGHT) %>%
  gather(sample, expression, G0.05:U0.3) %>%
  separate(sample, c("nutrient", "rate"), sep = 1, convert = TRUE)

cleaned_data %>%
  filter(BP == "leucine biosynthesis") %>%
  ggplot(aes(rate, expression, color = nutrient)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~name)
			  </code>
            </pre>
		  </section>

		  <!-- Analyses supplémentaires sur le même jeu de données -->
		  <section>
			<h2>Modeling gene expression with broom</h2>
			<h3>a case study in tidy analysis</h3>
            <a href="http://varianceexplained.org/r/tidy-genomics-broom/">
              <img width="200"
                   data-src="http://varianceexplained.org/images/david_robinson_picture2.jpg"
                   >
            </a>
            <p>par David Robinson</br><a href="http://varianceexplained.org/r/tidy-genomics-broom/">analyses supplémentaires sur le même jeu de données</a></p>
		  </section>

		  <!-- en chantier -->
		  <section>
			<h2>section en chantier</h2>
		  </section>

		  <!-- ggbio -->
		  <section>
			<h2>ggbio</h2>
			<br>
            <a href="http://www.tengfei.name/ggbio/">
			  <img width="720"
                   data-src="https://raw.githubusercontent.com/tengfei/ggbio/gh-pages/images/interval.png"
                   alt="visualization of genomic data with ggbio"
                   >
            </a>
		  </section>

		  <!-- ggtree -->
		  <section>
			<h2>ggtree</h2>
			<br>
            <a href="http://bioconductor.org/packages/release/bioc/html/ggtree.html">
			  <img width="720"
                   data-src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/master/Bioconductor/ggtree/2016-ggtree-phylip_files/figure-markdown_strict/unnamed-chunk-2-1.png"
                   alt="ggtree example with alignment"
                   >
            </a>
		  </section>

		  <!-- TARA results -->
		  <section>
			<h2>TARA Oceans</h2>
			<p>relation entre fréquence, dispersion et diversité des taxons</p>
			<img
               data-src="images/cosmopolites_vs_specialists_47_station.png"
               alt="TARA data, relation entre fréquence et dispersion des taxons"
               >
		  </section>

					<!-- GG_theme -->
				<section>
					<h2>Cosmétique: gg_theme()<h2>

				</section>

				<section>
					<h2>Exporter sa figure<h2>


				</section>


				<!-- Spatial and temporal datasets -->
				<section>
					<h3>Gerer des données temporelles/spatiales </h3>
					<p>
						Données d'abondances de cullicoides sur des piègages en France métropolitaine sur plusieurs années.
					</p>
							<img width = "50%" data-src="images/c_imicola_femelle_reference.jpg" alt="imicola" >

				</section>
				<section>
					<p>
					Format csv -> "comma separated value" 
					</p>
					<p>
						Les champs sont séparés par le charactère <strong>"|"</strong>
					</p>
							<img width = "100%" data-src="images/rawdataOcapi.png" alt="dataset" >

				</section>
				<section>
					<pre><code class="R" data-trim contenteditable>
data = read.table("uniqDataOcapi.csv",sep="|",header=T, na.strings = "NA",\
 fill=TRUE, strip.white=TRUE, blank.lines.skip = TRUE)
str(data)


					</code></pre>
					<pre><code class="R" data-trim contenteditable>
'data.frame':   54647 obs. of  12 variables:
$ ID_SITE     : Factor w/ 298 levels "01PL1","01PL2",..: 1 1 1 1 1 1 1 1 1 1 ...
$ LONG_IDENTIF: num  5.79 5.79 5.79 5.79 5.79 5.79 5.79 5.79 5.79 5.79 ...
$ LAT_IDENTIF : num  46.1 46.1 46.1 46.1 46.1 ...
$ DATEDEBUT   : Factor w/ 1133 levels "10/10/2008","10/10/2011",..: 12 26 51 69 69 69 69 158 170 203 ...
$ ESPECE      : Factor w/ 89 levels "abchazicus","achrayi",..: 15 15 53 44 53 54 65 15 53 15 ...
$ NBMALES     : int  0 0 0 0 0 1 0 0 0 0 ...
$ NBFEMELLES  : int  0 0 4 11 226 0 14 0 2 0 ...
$ NBPARE      : int  0 0 4 6 192 0 14 0 1 0 ...
$ NBNULLIPARE : int  0 0 0 5 34 0 0 0 1 0 ...
$ NBGORGEES   : int  0 0 0 0 0 0 0 0 0 0 ...
$ NBINDIV     : int  0 0 4 11 226 1 14 0 2 0 ...
$ NBINDETER   : int  0 0 0 0 0 0 0 0 0 0 ...
					</code></pre>
				
				</section>


				<!-- Data cleaning part  -->
			<section>
				 
				<section>
					<pre><code class="R" data-trim contenteditable>
 data %>% tbl_df  %>% View
					</code></pre>
				<img width = "100%" data-src="images/Ocapi.png" alt="dataset" >
				
				</section>
				
				<section>
					<pre><code class="R" data-trim contenteditable>
data %>% tbl_df %>% filter(., NBINDIV>0) %>% na.omit %>% View
					</code></pre>
				<img width = "100%" data-src="images/Ocapi1.png" alt="dataset" >
				
				
				</section>
				
				<section>
					<pre><code class="R" data-trim contenteditable>
data %>% tbl_df %>% filter(., NBINDIV>0) %>% na.omit %>% \
mutate(.,lat = as.numeric(LAT_IDENTIF)) %>% \
mutate(.,long = as.numeric(LONG_IDENTIF)) %>%  \
mutate(.,date = as.Date(DATEDEBUT,"%m/%d/%Y")) \ 
mutate(.,SexRatio = NBFEMELLES/(NBMALES+NBFEMELLES)) %>% View 
					</code></pre>

				<img width = "100%" data-src="images/Ocapi3.png" alt="dataset" >
				</section>

				
				<section>
					<pre><code class="R" data-trim contenteditable>
data %>% tbl_df %>% filter(., NBINDIV>0) %>% na.omit %>% \
mutate(.,lat = as.numeric(LAT_IDENTIF)) %>% \
mutate(.,long = as.numeric(LONG_IDENTIF)) %>%  \
mutate(.,date = as.Date(DATEDEBUT,"%m/%d/%Y")) \   
mutate(.,SexRatio = NBFEMELLES/(NBMALES+NBFEMELLES)) \ 
separate(.,date,c("year","month","day")) %>% View
					</code></pre>

				<img width = "100%" data-src="images/Ocapi4.png" alt="dataset" >
				</section>

				
				<section>
					<pre><code class="R" data-trim contenteditable>
data %>% tbl_df %>% filter(., NBINDIV>0) %>% na.omit %>% \
mutate(.,lat = as.numeric(LAT_IDENTIF)) %>% \
mutate(.,long = as.numeric(LONG_IDENTIF)) %>%  \
mutate(.,date = as.Date(DATEDEBUT,"%m/%d/%Y")) 
mutate(.,SexRatio = NBFEMELLES/(NBMALES+NBFEMELLES)) \ 
separate(.,date,c("year","month","day")) \
select(.,ID_SITE,long,lat,year,month,day,ESPECE,SexRatio,NBINDIV,NBMALES,NBFEMELLES,) %>% View
					</code></pre>

				<img width = "100%" data-src="images/Ocapi5.png" alt="dataset" >
				</section>
				
				<section>
					<pre><code class="R" data-trim contenteditable>
data %>% tbl_df %>% filter(., NBINDIV>0) %>% na.omit %>% \
mutate(.,lat = as.numeric(LAT_IDENTIF)) %>% \
mutate(.,long = as.numeric(LONG_IDENTIF)) %>%  \
mutate(.,date = as.Date(DATEDEBUT,"%m/%d/%Y")) 
mutate(.,SexRatio = NBFEMELLES/(NBMALES+NBFEMELLES)) \ 
separate(.,date,c("year","month","day")) \
select(.,ID_SITE,long,lat,year,month,day,ESPECE,SexRatio,NBINDIV,NBMALES,NBFEMELLES,) \
gather(.,"variable","value",NBINDIV,SexRatio,NBMALES,NBFEMELLES) %>% View  
					</code></pre>

				<img width = "100%" data-src="images/Ocapi6.png" alt="dataset" >
				</section>
			</section>

				
				<section>
					<pre><code class="R" data-trim contenteditable>
td = data %>% tbl_df %>% filter(., NBINDIV>0) %>% na.omit %>% \
mutate(.,lat = as.numeric(LAT_IDENTIF)) %>% \
mutate(.,long = as.numeric(LONG_IDENTIF)) %>%  \
mutate(.,date = as.Date(DATEDEBUT,"%m/%d/%Y")) 
mutate(.,SexRatio = NBFEMELLES/(NBMALES+NBFEMELLES)) \ 
separate(.,date,c("year","month","day")) \
select(.,ID_SITE,long,lat,year,month,day,ESPECE,SexRatio,NBINDIV,NBMALES,NBFEMELLES,) \
gather(.,"variable","value",NBINDIV,SexRatio,NBMALES,NBFEMELLES) 
					</code></pre>

					<pre><code class="R" data-trim contenteditable>
> td
Source: local data frame [193,332 x 9]

   ID_SITE  long   lat  year month   day             ESPECE variable value
    (fctr) (dbl) (dbl) (chr) (chr) (chr)             (fctr)   (fctr) (dbl)
1    01PL1  5.79 46.06  2009    10    26 obsoletus/scoticus  NBINDIV     4
2    01PL1  5.79 46.06  2009    10    05          lupicaris  NBINDIV    11
3    01PL1  5.79 46.06  2009    10    05 obsoletus/scoticus  NBINDIV   226
4    01PL1  5.79 46.06  2009    10    05     obsoletus s.st  NBINDIV     1
5    01PL1  5.79 46.06  2009    10    05          pulicaris  NBINDIV    14
6    01PL1  5.79 46.06  2009    11    23 obsoletus/scoticus  NBINDIV     2
7    01PL1  5.79 46.06  2009    04    14         chiopterus  NBINDIV    14
8    01PL1  5.79 46.06  2009    04    14          newsteadi  NBINDIV     1
9    01PL1  5.79 46.06  2009    04    14 obsoletus/scoticus  NBINDIV    58
10   01PL1  5.79 46.06  2009    04    14     obsoletus s.st  NBINDIV     1
..     ...   ...   ...   ...   ...   ...                ...      ...   ...


					</code></pre>
				</section>

			
				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>

td = data %>% tbl_df %>% filter(., NBINDIV>0) %>% na.omit %>% \
mutate(.,lat = as.numeric(LAT_IDENTIF)) %>% \
mutate(.,long = as.numeric(LONG_IDENTIF)) %>%  \
mutate(.,date = as.Date(DATEDEBUT,"%m/%d/%Y")) 
mutate(.,SexRatio = NBFEMELLES/(NBMALES+NBFEMELLES)) \ 
<mark>mutate(.,datesp = date)</mark>
separate(.,<mark>datesp</mark>,c("year","month","day")) \
select(.,ID_SITE,long,lat,year,month,day,ESPECE,SexRatio,NBINDIV,NBMALES,NBFEMELLES,) \
gather(.,"variable","value",NBINDIV,SexRatio,NBMALES,NBFEMELLES) 
					</code></pre>

					<pre><code class="R" data-trim contenteditable>

> td
Source: local data frame [193,332 x 10]

   ID_SITE  long   lat       date  year month   day             ESPECE variable
    (fctr) (dbl) (dbl)     (date) (chr) (chr) (chr)             (fctr)   (fctr)
1    01PL1  5.79 46.06 2009-10-26  2009    10    26 obsoletus/scoticus  NBINDIV
2    01PL1  5.79 46.06 2009-10-05  2009    10    05          lupicaris  NBINDIV
3    01PL1  5.79 46.06 2009-10-05  2009    10    05 obsoletus/scoticus  NBINDIV
4    01PL1  5.79 46.06 2009-10-05  2009    10    05     obsoletus s.st  NBINDIV
5    01PL1  5.79 46.06 2009-10-05  2009    10    05          pulicaris  NBINDIV
6    01PL1  5.79 46.06 2009-11-23  2009    11    23 obsoletus/scoticus  NBINDIV
7    01PL1  5.79 46.06 2009-04-14  2009    04    14         chiopterus  NBINDIV
8    01PL1  5.79 46.06 2009-04-14  2009    04    14          newsteadi  NBINDIV
9    01PL1  5.79 46.06 2009-04-14  2009    04    14 obsoletus/scoticus  NBINDIV
10   01PL1  5.79 46.06 2009-04-14  2009    04    14     obsoletus s.st  NBINDIV
..     ...   ...   ...        ...   ...   ...   ...                ...      ...
Variables not shown: value (dbl)


					</code></pre>
				</section>

				<section>




				</section>








				<!-- Spatial and temporal datasets -->


				<section>
					<h2>graphique interactif avec plotly</h2>
					<iframe src="Fit_piS.html" width="800" height = "500" frameborder = 0>
					</iframe>
				</section>



				<section>
				  <h2>Tabular Tables</h2>
				  <table>
					<thead>
					  <tr>
						<th>Item</th>
						<th>Value</th>
						<th>Quantity</th>
					  </tr>
					</thead>
					<tbody>
					  <tr>
						<td>Apples</td>
						<td>$1</td>
						<td>7</td>
					  </tr>
					  <tr>
						<td>Lemonade</td>
						<td>$2</td>
						<td>18</td>
					  </tr>
					  <tr>
						<td>Bread</td>
						<td>$3</td>
						<td>2</td>
					  </tr>
					</tbody>
				  </table>
				</section>

                <!-- Conclusion -->
				<section style="text-align: left;">
				  <h1>THE END</h1>
				  <p>
					- <a href="http://slides.com">Try the online editor</a> <br>
					- <a href="https://github.com/hakimel/reveal.js">Source code &amp; documentation</a>
				  </p>
				</section>

			</div>
		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.js"></script>

		<script>

			// Full list of configuration options available at:
			// https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				controls: true,
				progress: true,
				history: true,
				center: true,

				transition: 'slide', // none/fade/slide/convex/concave/zoom

				// Optional reveal.js plugins
				dependencies: [
					{ src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
					{ src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
					{ src: 'plugin/zoom-js/zoom.js', async: true },
					{ src: 'plugin/notes/notes.js', async: true }
				]
			});

		</script>

	</body>
</html>
